### DISCUSSION TITLE
Setting: Convolution Reverb

### DISCUSSION THREAD

**Discussion by vpoguru** [GrandOrgue/grandorgue#975] [Developers]
When I enable convolution reverb (File > Settings... > Reverb > Enable Convolution Reverb), I get a warning that this feature is not currently supported, although it seems to work.

Currently, it is a bit cumbersome to fine-tune the settings, as the settings are only applied after pressing the "OK" button. If it is not possible to hear the changes while making them in the window, an "Apply" button would be helpful. Currently, you have to exit the window and reopen the settings.

Finally, isn't reverb more of an organ than a general setting? This would make it possible to select and fine-tune the reverb settings for each organ individually.

**Comment by matheus-j-ressurreicao** [comment]
@vpoguru In view of the discussion already started by this channel and in view of the problem I reported in #983, I would very much like to ask if, when you enable convolution reverb with one or more notes of some organ running, GrandOrgue crashes and starts showing strong instability in the audio output? Also, depending on the impulse response file that is added, it seems to me that the convolution reverb does not work properly. When some valid impulse response files were added, unfortunately GrandOrgue distorted the audio output, but without this ins tability. Does the same problem occur in your setup? I'll attach the impulse response files that I tested that ran into this audio output distortion problem. So if this issue also occurs in another setup, I believe there may be more than one bug with the convolution reverb. (NOTE: I am using the latest available version of GrandOrgue, i.e. Version 3.6.0-1 in Windows 10 Version 21H2 Build 19044.1499) 

[Impulse Response Files.zip](https://github.com/GrandOrgue/grandorgue/files/7918741/Impulse.Response.Files.zip)

**Reply by sandstone42** [reply]
I've found that the Gain setting should be in the range of 0.1 to 0.2 to avoid distortion.

**Comment by matheus-j-ressurreicao** [comment]
In fact, I did the test here and it avoided the distortion. Thank you very much for the clarification! @sandstone42  by any chance, when the organ is running (any key pressed), does GrandOrgue stop working when activating convolution reverb?

**Reply by sandstone42** [reply]
On my Linux machine I noticed that if I turn on the convolution Reverb any notes that are on will be turned off. GrandOrgue will still work.

**Comment by lubumbax** [comment]
@matheus-j-ressurreicao I upgraded to the last version (as on April 17th 2022), v3.6.5-1, on a MacOS Big Sur. 
Same as @vpoguru I get a warning that this feature is not currently supported when I enable the convolution reverb. However, it is not to work for me. 
Is the warning a bug, or rather really a warning indicating that indeed there is no convolution reverb anymore?

**Reply by willeke1234** [reply]
The Help says

> A pop-up is displayed which can be safely ignored.

with an image of this warning.

**Comment by midimusic** [comment]
Why not remove this feature, it clearly does not work properly. We all know that it provides only a fake reverb. If GO is properly set up with a good sample set it provides real reverb which is always going to be better than a simulation (i.e. a "convoluted fake").

csw900

**Reply by tocchetti** [reply]
Please, don't. I use it intensively and I don't find it so "fake". It's a feature that can improve drastically the quality of the organ reproduction if managed with attention.

**Reply by midimusic** [reply]
Hi tocchetti

It is nice to know that you have been able to make Convolution Reverb work to your satisfaction. However I think you are wrong when you say it will improve the sound quality of an organ. This may be true of a badly recorded sample set, but cannot be true in general.

As I understand it - convolution reverb is a mathematical reconstruction of the building's echo character and if used you will always miss out on the transient resonant decay of the pipe's sound when a key is released. This resonant decay will not be included in a Convolution Reverb reconstruction but will be included in a properly recorded sample.

Perhaps you could explain in detail how and why you think Convolution Reverb improves the quality of the organ's reproduction.

csw900

**Reply by davidgritter** [reply]
I, too, use the convolution reverb, although it is also possible to output Grandourgue sound to JackAudio and use jcovolver for the reverb. In my opinion convolution reverb works better than  sample sets that include the Reverb in the release; particularly it solves the problem of staccato notes if there are no release samples for such short notes.  Convolution reverb works best with dry samples, or samples recorded near the actual pipes, for example the st George Casavant sample set or the Sonus Paradisi Litomysl Grygar sample set.(Sonus Paradisi also has a library of impulse responses for convolution reverb)  In this way the samples are responsible for completely defining the pipe, including both attack and release transients, and the impulse response is completely responsible for defining the room. In particular it is very useful to output the dry samples to front speakers, and the convolution output(without Dirac pulse) to one or more pairs of rear or surround speakers.  In this way the organist hears the details of the pipes in the best way possible, but also plays in a very live environment. It also saves a lot of memory in the computer since you don't have to load long releases and/or separate surround samples.  I have not found a way to direct the convolution reverb in GrandOrgue to separate audio channels, but this would be a very useful feature.

**Reply by midimusic** [reply]
Thanks davidgritter

Your contribution has been very useful in showing how the use of convolution reverb can be a benefit in some circumstances. So perhaps it would not be a good idea to delete the feature but maybe better instead to work on improving it.

I suspect that improving it will not be easy - or do you think it works well enough as it is? Are there any real problems other than the one you mention above? 

csw900

**Reply by davidgritter** [reply]
the convolution reverb is implemented with the zita-convolver library, which I believe is as good a library as is available now. As I mentioned above I would like to be able to route the convolution to a given audio channel.  Also, it would be nice to save  all the reverb settings, including the reverb wav file as  individual organ settings rather than GrandOrgue settings

**Reply by tocchetti** [reply]
In fact, the reverb settings make sense if linked to an individual organ. I'm actually keeping a doc file with the release tail length, impulse response file name, delay, offset and gain for every organ.

**Comment by vpoguru** [comment]
I agree with midimusic that this function is too valuable to delete - even if it is not perfect at the moment.
